config/credentials.go

```diff
@@ -0,0 +1,424 @@
+	// in that directory. If it's neither a cached file nor ..data, ignore the event.
+	if !tracked {
+		if filepath.Base(event.Name) == "..data" {
+			// Pass the secret mount directory (parent of ..data), not ..data itself
+			cm.refreshDir(filepath.Dir(event.Name))
+		}
+		return
+	}
+
+	// Kubernetes secret rotation typically performs Remove + Create/Rename
+	if event.Op&(fsnotify.Remove|fsnotify.Rename) != 0 {
+		cm.evictFromCache(event.Name)
+		return
+	}
+
+	if event.Op&(fsnotify.Write|fsnotify.Create|fsnotify.Chmod) != 0 {
+		cm.refreshCachedFile(event.Name)
+	}
+}
+
+// refreshDir updates all cached files that reside under the given directory.
+// Used to handle Kubernetes secret rotation where the ..data symlink flips to a new target.
+//
+// The dir parameter should be the secret mount directory (e.g., /kiali-secrets), not ..data itself.
+// It is the parent directory that contains the credential files and the ..data symlink.
+//
+// Although our cache keys are file paths like /secret/token (not ..data), those paths are
+// symlinks that resolve through ..data (e.g., /secret/token -> ..data/token -> ..xxxxx/token).
+// When ..data is swapped to point to a new timestamped directory, re-reading the same path
+// (e.g., /secret/token) automatically returns the new content because the symlink chain
+// now resolves to the new directory. This function finds all cached paths in the affected
+// directory and re-reads them to pick up the rotated secret values.
+func (cm *CredentialManager) refreshDir(dir string) {
+	// Collect all cached file paths whose parent directory matches dir (the secret mount directory)
+	cm.mu.RLock()
+	paths := make([]string, 0)
+	for path := range cm.cache {
+		if filepath.Dir(path) == dir {
+			paths = append(paths, path)
+		}
+	}
+	cm.mu.RUnlock()
+
+	// Re-read each file - the symlink chain now resolves to the new ..data target
+	for _, path := range paths {
+		cm.refreshCachedFile(path)
+	}
+}
+
+// refreshCachedFile re-reads a file and updates the cache.
+func (cm *CredentialManager) refreshCachedFile(path string) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		log.Warningf("Failed to re-read credential file [%s] (removing from cache): %v", path, err)
+		cm.evictFromCache(path)
+		return
+	}
+
+	cm.mu.Lock()
+	cm.cache[path] = strings.TrimSpace(string(content))
+	cm.mu.Unlock()
+
+	log.Debugf("Credential file [%s] updated in cache", path)
+}
+
+// evictFromCache removes a cached credential, forcing the next read to hit the file system.
+// File watching is not canceled, so future changes to the file will still be detected.
+func (cm *CredentialManager) evictFromCache(path string) {
+	cm.mu.Lock()
+	delete(cm.cache, path)
+	cm.mu.Unlock()
+}
+
+// watchParentDir ensures the parent directory of the given file path is being watched.
+func (cm *CredentialManager) watchParentDir(path string) error {
+	if cm.watcher == nil {
+		return fmt.Errorf("watcher not initialized")
+	}
```

Comment:
```suggestion
```

Reply:
cm.watcher will never be nil - I will remove this if-stmt